assembly/src/assembler/dead_code_elimination.rs

```diff
@@ -0,0 +1,155 @@
+use alloc::{
+    collections::{BTreeMap, BTreeSet},
+    vec::Vec,
+};
+
+use vm_core::mast::{MastForest, MastNode, MastNodeId};
+
+/// Returns a `MastForest` where all nodes that are unreachable from all procedures are removed.
```

Comment:
nit: should this say "... where all nodes that are unreachable from **any** procedures are removed."?

Reply:
This would imply that if one procedure doesn't visit a node, then we would remove that node. Instead, what we are saying here is: if no procedure visit a given node, then we remove that node.

Changed "all nodes" to only "nodes" to make the sentence slightly less heavy to parse.